src/dnaio/__init__.py

```diff
@@ -122,21 +131,51 @@ def open(
        A subclass of `SingleEndReader`, `PairedEndReader`, `SingleEndWriter` or
        `PairedEndWriter`.
     """
-    if mode not in ("r", "rb", "w", "a"):
-        raise ValueError("Mode must be 'r', 'rb', 'w' or 'a'")
-    elif interleaved and file2 is not None:
-        raise ValueError("When interleaved is True, file2 must be None")
-    elif interleaved or file2 is not None:
+    if files and (file1 is not None):
+        raise ValueError(
+            "the file1 argument cannot be used together with unnamed files"
+        )
+    elif len(files) > 1 and file2 is not None:
+        raise ValueError(
+            "the file2 argument cannot be used together with more than one "
+            "unnamed file"
+        )
+    elif file1 is not None and file2 is not None and files:
+        raise ValueError(
+            "file1 and file2 arguments cannot be used together with unnamed files"
+        )
+    elif file1 is not None and file2 is not None:
+        files = (file1, file2)
+    elif file2 is not None and len(files) == 1:
+        files = (files[0], file2)
+
+    if len(files) > 1 and interleaved:
+        raise ValueError(
+            "When interleaved is True only one file must be specified as input."
+        )
+    elif mode not in ("r", "w", "a"):
+        raise ValueError("Mode must be 'r', 'w' or 'a'")
+
+    if interleaved or len(files) == 2:
         return _open_paired(
-            file1,
-            file2=file2,
+            files[0],
+            file2=None if interleaved else files[1],
```

Comment:
I see some redundancy here in open paired. Either you give one or two files. One means interleaved, two means not. The interleaved parameter is this not necessary for the internal _open_paired function. It will be much easier simply to use `_open_paired(*files, **other_kwargs` since we have  already established above that the correct conditions are met.

Reply:
I agree, this could be simpler.